### PR TITLE
SEQNG-751: Send the observer when starting a queue

### PR DESCRIPTION
**Pull request by cquiroz**
The observer should be set when starting a queue. This PR implements this partially as the backend is not complete yet. There are a couple of edge cases that need work but this is enough to make a release and let users test the queue

Note that I'm changin some `Map` to `SortedMap` to avoid using the unsafe Monocle lenses

https://monosnap.com/file/oOUEz2jAfa6AhrcEgSaeGx5kZFXoGI#

### CODE REVIEW

File: modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueueView.scala [hunk 1]
```diff
@@ -7,17 +7,24 @@ import cats.Eq
 import cats.implicits._
 import gem.Observation
 import monocle.macros.Lenses
-import seqexec.model.enum.{BatchCommandState, BatchExecState}
+import seqexec.model.enum.BatchCommandState
+import seqexec.model.enum.BatchExecState
```
Comment (sraaphorst) [inline modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueueView.scala]: Why separate these? Is this a style thing?

Comment (cquiroz) [reply modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueueView.scala]: yes, otherwise I easily have to split the import into several lines, this is handled automatically

File: modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueueView.scala [hunk 2]
```diff
@@ -7,17 +7,24 @@ import cats.Eq
...
 
 @Lenses
 final case class ExecutionQueueView(id:        QueueId,
                                     name:      String,
                                     cmdState:  BatchCommandState,
                                     execState: BatchExecState,
-                                    queue:     List[Observation.Id])
+                                    queue:     List[Observation.Id]) {
+  val observer: Option[Observer] = cmdState match {
```
Comment (cquiroz) [inline modules/seqexec/model/src/main/scala/seqexec/model/ExecutionQueueView.scala]: I'm not happy about this but it will do for now. Ideally we'd have the observer in the model for real

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/HeadersSideBar.scala [hunk 3]
```diff
@@ -27,26 +44,38 @@ import scala.concurrent.duration._
 object HeadersSideBar {
   final case class Props(model: ModelProxy[HeaderSideBarFocus]) {
     def canOperate: Boolean = model().status.canOperate
-    def sequenceObserver: Either[Observer, SequenceObserverFocus] = model().observer
+    def selectedObserver
+      : Either[Observer, Either[DayCalObserverFocus, SequenceObserverFocus]] =
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/HeadersSideBar.scala]: Ugly type to have the options:
* Default observer
* Daycal observer
* instrument observer

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/SequencesOnDisplay.scala [hunk 4]
```diff
@@ -57,6 +64,19 @@ final case class SequencesOnDisplay(tabs: Zipper[SeqexecTab]) {
       case InstrumentSequenceTab(_, Some(curr), _, _, _, _) => curr.id
     }
 
+  /**
+    * Update the observer for the cal tab
+    */
+  def updateCalTabObserver(o: Observer): SequencesOnDisplay = {
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/SequencesOnDisplay.scala]: We update the local Cal queue observer here

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/SequencesOnDisplay.scala [hunk 5]
```diff
@@ -265,14 +300,23 @@ final case class SequencesOnDisplay(tabs: Zipper[SeqexecTab]) {
...
     */
-  def selectedOperator: Option[SequenceObserverFocus] =
-    SequencesOnDisplay.focusSequence.getOption(this).collect {
-      case InstrumentSequenceTab(_, Some(s), _, _, _, _) =>
-        SequenceObserverFocus(s.metadata.instrument,
-                              s.id,
-                              s.allStepsDone,
-                              s.metadata.observer)
-    }
+  def selectedObserver
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/SequencesOnDisplay.scala]: Return the observer for the selected tab

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/HeadersSideBar.scala [hunk 6]
```diff
@@ -106,71 +152,108 @@ object HeadersSideBar {
...
-          f.backend.updateStateOb(o.value.some, Callback.empty)
-      })
+    .componentWillMount(f =>
+      f.backend.$.props >>= { p =>
+        p.model()
+          .operator
+          .map(op => f.backend.updateStateOp(op.value.some, Callback.empty))
+          .getOrEmpty *>
+          (p.selectedObserver match {
+            case Right(Right(a)) =>
```
Comment (sraaphorst) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/HeadersSideBar.scala]: 😆 
The dreaded nested `Either`.

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/tabs.scala [hunk 7]
```diff
@@ -81,24 +90,34 @@ object SeqexecTab {
...
       case i: InstrumentSequenceTab => i
     }(identity)
 }
 
-final case class CalibrationQueueTab(tableState: TableState[StepsTable.TableColumn], state: BatchExecState) extends SeqexecTab {
+@Lenses
+final case class CalibrationQueueTab(
+  tableState: TableState[StepsTable.TableColumn],
+  state:      BatchExecState,
+  observer:   Option[Observer])
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/tabs.scala]: This should be on the core model, but lacking that we need to keep it on the client model, hence this field
